cypress/integration/smoke/gateways/delete.js

```diff
@@ -0,0 +1,47 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { defineSmokeTest } from '../utils'
+
+const gatewayDelete = defineSmokeTest('succeeds deleting a gateway', () => {
+  const user = {
+    ids: { user_id: 'gtw-delete-test-user' },
+    primary_email_address: 'test-user@example.com',
+    password: 'ABCDefg123!',
+    password_confirm: 'ABCDefg123!',
+    email: 'gtw-delete-test-user@example.com',
+  }
+  cy.createUser(user)
+  const gateway = {
+    ids: { gateway_id: 'gtw-test-delete' },
+  }
+  cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
+  cy.createGateway(gateway, user.ids.user_id)
+  cy.visit(
+    `${Cypress.config('consoleRootPath')}/gateways/${gateway.ids.gateway_id}/general-settings`,
+  )
+  cy.findByRole('button', { name: /Delete gateway/ }).click()
+  cy.findByTestId('modal-window').within(() => {
+    cy.findByRole('button', { name: /Delete gateway/ }).click()
+  })
+
+  cy.findByTestId('full-error-view').should('not.exist')
+
+  cy.location('pathname').should('eq', `${Cypress.config('consoleRootPath')}/gateways`)
+  cy.get('tbody').within(() => {
+    cy.findByRole('cell', { name: gateway.ids.gateway_id }).should('not.exist')
+  })
```

Comment:
You might want to check whether the gateway was visible in the table before deleting it as well.

Reply:
> See my other commend regarding the creation form.
> Regarding the tests: make sure that the checks stay closely scoped to the test-case. We shouldn't test every last pixel simply because we can. We need to ensure that tests do not lock us into one specific implementation more than necessary.
> It's not always trivial to do it and it shouldn't block you too much from building test cases, but please keep it in mind when writing tests.

I agree completely, but I am really not sure what to test for then. I think testing these forms this thoroughly (which also concerns the test you wrote for create application view) should be scoped to unit tests we already have, and then the tests should indeed be pixel by pixel, and changed once something is updated in the text itself.

I like the way we use smoke tests, which exactly cover the use cases with concrete actions and a few UI checks to be sure the thing we test for really happened, but regarding the view tests, I am not so sure what are the tests about. cc @kschiffer

Reply:
Addressing the code comments I've made on the tests is completely sufficient here. I'm just saying that e.g. checking on the exact text content of a description is too much of an irrelevant detail, while checking for unset values makes sense since it is relevant to the test-case (of checking correct form behavior).

I agree that the exact separation between our smoke and component tests are not straightforward and tend to overlap.

Reply:
Regarding the original comment here. I think since this is a smoke test, it's fine to use a wider scope of checks that resemble more of a complete user story. Adding another check here to verify that the gateway was there in the first place is a cheap way to improve the expressiveness of this test.